packages/cli/src/lifecycles/serve.js

```diff
@@ -69,7 +69,9 @@ function getProdServer(compilation) {
     const { outputDir } = compilation.context;
 
     if (ctx.request.url.endsWith('/')) {
-      ctx.redirect(`http://localhost:8080${ctx.request.url}index.html`);
+      const contents = await fsp.readFile(path.join(outputDir, ctx.request.url, 'index.html'), 'utf-8');
```

Comment:
Does combining the `/` and `.html` tests not work well together? 
```js
if (ctx.request.url.endsWith('/')) {
  const contents = await fsp.readFile(path.join(outputDir, ctx.request.url, 'index.html'), 'utf-8');
  ctx.set('Content-Type', 'text/html');
  ctx.body = contents;
}
```

Or I suppose it could also make sense to keep `.html` as a stand alone handler as well?

Reply:
right I think need to keep it in case?

Reply:
> right I think need to keep it in case?

in case of what though?  I wasn't sure if you had a case in mind, so the question was mostly for you.  ;)

What cases do we have of needing to process _.html_ differently?  I could at least see treating both with the same logic perhaps?
```js
if (ctx.request.url.endsWith('/') || ctx.request.url.endsWith('.html')) {
  const barePath = url.endsWith('/') ? '' : ctx.request.url;
  const contents = await fsp.readFile(path.join(outputDir, barePath, 'index.html'), 'utf-8');

  ctx.set('Content-Type', 'text/html');
  ctx.body = contents;
}
```

Mostly just spitballing at this point, so no big deal.

Reply:
lets go with that then. I don't have an immediate use case are we resolving all custom html files to their own routes and prettifying them ?  e.g. docs.html would just be /docs/ ? I don't know I havent tested that yet.

Reply:
yeah, give it a test.  At the end of the day they both _return_ the contents of an HTML file, it's just that we can do it simply without including a `redirect` in either case.